### ISSUE TITLE
How Do do I update Packery's layout?

### ISSUE THREAD

**Issue by digisocialnet** [t-k-f/vue-packery-plugin#3]
```
<div class="columns is-multiline is-variable is-5" v-packery='{itemSelector: ".packery-item", percentPosition: true, stagger: 30}'>
      <div v-packery-item v-for="(step, index) in steps" v-bind:key="step.id" @click="toggle(step.clicked)" :index="index" class="column packery-item">
        <article>
        <div class="notification step is-pink-outline is-bold">
          <figure v-if="step.image" class="image is-1by1">
          <img :src="step.image">
          </figure>
        <h4 class="title">
        {{step.heading}}
      </h4>
        <p>
         {{step.text}}
       </p>
       <p v-bind:key="step.id" v-show="step.clicked">
         {{step.fulltext}}
       </p>
        </div>
       </article>
      </div>
   </div>



    methods: {
                toggle (clicked, $event) {
 event.currentTarget.classList.toggle('expand')
    }
     }
```

How do I insert a update layout command to the page? The method resizes my box but doesn't update the rest of the layout.

**Comment by jan-thoma** [comment]
looks like an issue with non loaded images at the time the change event is emitted. If you know the image dimensions you can absolute position them in a container with padding top:

```
<span style="padding-top: 50%">
  <img src="some.jpg">
</span>
```

CSS:

```
span img {
  position:absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
```

Maybe i'm integrate imagesLoaded as an option.

**Comment by digisocialnet** [comment]
I'm also having problems destroying my instance on route change. I did add in the intrinsic ratio to my images.

**Comment by jan-thoma** [comment]
Can you provide an example? Proper destruction of instance was added at version 1.0.5

**Comment by digisocialnet** [comment]
Is this enough? My console is 1st, then index.vue and nuxt.config

nuxt-error.vue:48 TypeError: Cannot read property 'destroy' of undefined
    at unbind (packery.js:48)
    at callHook$1 (vue.runtime.esm.js:5851)
    at _update (vue.runtime.esm.js:5815)
    at updateDirectives (vue.runtime.esm.js:5757)
    at Array.unbindDirectives (vue.runtime.esm.js:5751)
    at invokeDestroyHook (vue.runtime.esm.js:5336)
    at invokeDestroyHook (vue.runtime.esm.js:5340)
    at invokeDestroyHook (vue.runtime.esm.js:5340)
    at invokeDestroyHook (vue.runtime.esm.js:5340)
    at VueComponent.patch [as __patch__] (vue.runtime.esm.js:5647)

```
<section>
  <div class="steps container is-fluid">

  <div class="columns is-multiline is-variable is-5" v-packery="{itemSelector: '.packery-item', percentPosition: true, stagger: 40}">
<div v-packery-item v-bind:data-size="steps.intro.size | booleanfilt" v-bind:key="steps.intro.heading" class="column packery-item">
        <article>
        <div class="notification intro is-light is-bold">
        <h4 class="title is-primary">
        {{steps.intro.heading}}
      </h4>
        <h5>
         {{steps.intro.text}}
      </h5>
        </div>

       </article>
</div>
      <div v-packery-item v-for="(stp, index) in steps.regimen" v-bind:data-clicked="stp.clicked | booleanfilt" v-bind:data-size="stp.size | booleanfilt" :index="index" v-bind:key="stp.id" class="column packery-item step">
        <article><nuxt-link :to="`/step/${stp.text}`">
        <div class="notification step is-pink-outline is-bold">
          <figure v-if="stp.image" class="image is-1by1">
          <img :src="stp.image">
          </figure>

                <div class="step-number">
         {{stp.text}}
      </div> <h4 class="title">{{stp.heading}} <transition name="slide">
       <span class="productname" v-if="stp.clicked">
         {{stp.productname}}
       </span>
</transition></h4>

        </div></nuxt-link>
       </article>
      </div>

   </div>
</div>
</section>
```

```
build: {
    vendor: [
      '~/plugins/vuepackeryplugin'
    ]
  },
plugins: [
    {src: '~/plugins/vuepackeryplugin', ssr: false},
]
```

**Comment by jan-thoma** [comment]
actually i mean a running example like jsfiddle.

**Comment by tophat1986** [comment]
DId you find a solution to this?  I'm also having the same issue and don't know how to instruct the layout command to this plugin to re-draw page on change of div size. :(

**Comment by jan-thoma** [comment]
Update to 1.2.1

```
import {packeryEvents} from 'vue-packery-plugin'

...

packeryEvents.$emit('layout', MY_PACKERY_EL)
```

**Comment by tophat1986** [comment]
Cool, many thanks for the updates!

However, being a beginner, I can't get it to work still for my situation.  I presume "MY_PACKERY_EL" is the element target div of the "v-packery-item" right?  This would invoke on the click of the HTMLDivElement itself I think to pass the $event into the nodes request. 

What I'm trying to do is have a button that triggers a change on ALL items in the v-packery list.  So I can't pass this event in as such.  Any ideas how to use this for a change on all the "v-packery-item" at once?  Will help if I do a reduced test case?

Thanks,

**Comment by jan-thoma** [comment]
No you need to emit the event to the v-packery container not the single items. if v-packery is the root element of your component use:

`this.$el`

else put a ref on the v-packery and use:

`this.$refs.myRef`

See: [https://vuejs.org/v2/api/#ref](https://vuejs.org/v2/api/#ref)

**Comment by tophat1986** [comment]
Thanks for pointing me in the right direction.  Had to use $refs in the end and now resolved. :)

**Comment by ssuess** [comment]
Not sure if something has changed since a year ago, but I am unable to trigger a change using 

`packeryEvents.$emit('layout', this.$refs.myPackeryContainer)`

I have verified my ref is correct (and output it to the console to be sure), and a new layout for my components WILL be triggered if I resize the window, so I know the code is working properly (if it could only be triggered on load). I have even tried setting a timeout to eliminate the possibility of any race condition, but that too fails. 

(And yes I have verified that `import {packeryEvents} from 'vue-packery-plugin'` is present)

Any suggestions? Thanks!
